lib/samples/navigate_route_with_rerouting/navigate_route_with_rerouting.dart

```diff
@@ -0,0 +1,587 @@
+                      ),
+                    ],
+                  ),
+                ),
+                Padding(
+                  padding: const EdgeInsets.fromLTRB(10, 0, 8, 8),
+                  child: Row(
+                    children: [
+                      Expanded(
+                        child: Text(
+                          'Time remaining: $_remainingTime',
+                          style: Theme.of(context).textTheme.labelMedium,
+                        ),
+                      ),
+                    ],
+                  ),
+                ),
+                Padding(
+                  padding: const EdgeInsets.fromLTRB(10, 0, 8, 24),
+                  child: Row(
+                    children: [
+                      Expanded(
+                        child: Text(
+                          'Next direction: $_nextDirection',
+                          style: Theme.of(context).textTheme.labelMedium,
+                          softWrap: true,
+                          maxLines: 2,
+                          overflow: TextOverflow.ellipsis,
+                        ),
+                      ),
+                    ],
+                  ),
+                ),
+              ],
+            ),
+            // Display a progress indicator and prevent interaction until state is ready.
+            LoadingIndicator(visible: !_ready),
+            // Display a progress indicator while the navigation data is loading.
+            Visibility(
+              visible: _setupDataAndInitNavigation,
+              child: const Center(
+                child: SizedBox(
+                  width: 300,
+                  child: Column(
+                    mainAxisAlignment: MainAxisAlignment.center,
+                    children: [
+                      LinearProgressIndicator(),
+                      Text('Downloading data...'),
+                    ],
+                  ),
+                ),
+              ),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  @override
+  void dispose() {
+    _flutterTts.stop();
+    _simulatedLocationDataSource?.stop();
+
+    super.dispose();
+  }
+
+  // Set up the map with a navigation basemap style.
+  void onMapViewReady() {
+    // Create a map with a navigation basemap style.
+    final map = ArcGISMap.withBasemapStyle(BasemapStyle.arcGISNavigation);
+    _mapViewController.arcGISMap = map;
+
+    
+
+    map.onLoadStatusChanged.listen((loadStatus) async{
+      if (loadStatus == LoadStatus.loaded) {
+        setState(() => _setupDataAndInitNavigation = true);
+      await initRouteTask();
+      setState(() => _setupDataAndInitNavigation = false);
+      }
+    });
+
+    setState(() => _ready = true);
+  }
+
+  // Initialize the route task, route parameters, and route result.
+  Future<void> initRouteTask() async {
+    final geodatabasePath = await _geodatabasePathFuture;
+    _simulatedLocationDataSource = await _simulatedLocationDataSourceFuture;
```

Comment:
`_simulatedLocationDataSource` isn't needed until later in the process, down in `initNavigation()`. We should move this to where  `_simulatedLocationDataSource` first gets used. That way, the Future has more time to complete.